realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -615,31 +717,47 @@ protected final void notifyCallbackError(ObjectServerError e) {
         // Combine error messages. If they have the same ErrorCode, it will be re-used, otherwise
         // we are forced to report back UNKNOWN as error code. The real error codes
         // will be always part of the exception message.
-        private ObjectServerError combineRealmErrors(ObjectServerError managementError, ObjectServerError
-                permissionError) {
+        private ObjectServerError combineRealmErrors(Map<String, ObjectServerError> errors) {
 
-            String errorMsg = combineErrorMessage(managementError, permissionError);
-            ErrorCode errorCode = (managementError.getErrorCode() == permissionError.getErrorCode())
-                    ? managementError.getErrorCode()
-                    : ErrorCode.UNKNOWN;
+            String errorMsg = combineErrorMessage(errors);
+            ErrorCode errorCode = combineErrorCodes(errors);
 
             return new ObjectServerError(errorCode, errorMsg);
         }
 
         // Combine the text based error message from two ObjectServerErrrors.
-        private String combineErrorMessage(ObjectServerError managementError,
-                                           ObjectServerError permissionError) {
-            StringBuilder errorMsg = new StringBuilder("Multiple errors occurred: ");
-            errorMsg.append('\n');
-            errorMsg.append("Management Realm:");
-            errorMsg.append('\n');
-            errorMsg.append(managementError.toString());
-            errorMsg.append('\n');
-            errorMsg.append("Permission Realm:");
-            errorMsg.append('\n');
-            errorMsg.append(permissionError.toString());
+        private String combineErrorMessage(Map<String, ObjectServerError> errors) {
+            boolean multipleErrors = errors.size() > 1;
+            StringBuilder errorMsg = new StringBuilder(multipleErrors ? "Multiple errors occurred: " : "Error occurred in Realm: ");
+            for (Map.Entry<String, ObjectServerError> entry : errors.entrySet()) {
+                errorMsg.append('\n');
+                errorMsg.append(entry.getKey());
+                errorMsg.append('\n');
+                errorMsg.append(entry.getValue().toString());
+            }
             return errorMsg.toString();
         }
+
+        private ErrorCode combineErrorCodes(Map<String, ObjectServerError> errors) {
+            ErrorCode finalErrorCode = null;
+            for (ObjectServerError error : errors.values()) {
```

Comment:
That doesn't catch the case where all Realms failed with the same error code. Which is actually likely if someone was wrong with the server.